### ISSUE TITLE
question about disc_loss

### ISSUE THREAD

**Issue by cp15512516229** [AILabDsUnipi/MultiModalIL#1]
"I want to know why this code uses generator_loss = discrim_criterion(g_o, ones((states.shape[0], 1), device=device)) instead of generator_loss = discrim_criterion(g_o, zeros((states.shape[0], 1), device=device)). Can you explain the reasoning behind this?"
"Is this project available?"

**Comment by AILabDsUnipi** [comment]
Hello, 
We followed the standard GAIL update process: The expert samples are labeled as real (0), while the generated ones are labeled as fake (1). The generator wants to fool the discriminator into classifying its generated samples as real. Using ones(), we encourage the discriminator to assign high probability to generated samples . If we used zeros(), the generator would help the discriminator classify its outputs as fake, which is the opposite of its goal.

**Comment by cp15512516229** [comment]
> Hello, We followed the standard GAIL update process: The expert samples are labeled as real (0), while the generated ones are labeled as fake (1). The generator wants to fool the discriminator into classifying its generated samples as real. Using ones(), we encourage the discriminator to assign high probability to generated samples . If we used zeros(), the generator would help the discriminator classify its outputs as fake, which is the opposite of its goal.

Thank you for your answer. In addition, I would like to know whether this project has been implemented. When I continue to develop the project, I find that there are many small errors in the project.

**Comment by cspatharis** [comment]
Hello again. This project has finished, but you can let me know of your suggestions.

**Comment by cp15512516229** [comment]
Hello ！ About the first question .Your statement contains fundamental errors. In the standard GAIL framework, expert demonstrations should be labeled as real data (label 1) and generated samples as fake data (label 0), which aligns perfectly with the adversarial logic of GANs. The generator aims to deceive the discriminator by maximizing its confidence in generated samples (pushing them toward 1), ​not by directly using ones() labels to enforce this assignment. Labeling expert samples as 0 would invert the discriminator's learning objective, contradict the generator's optimization direction in adversarial training, and ultimately undermine the core mechanism of imitation learning—progressively aligning the generated policy's distribution with the expert's. This inversion fatally disrupts the adversarial equilibrium required for effective policy imitation.

**Comment by cp15512516229** [comment]
> Hello again. This project has finished, but you can let me know of yHello ！ About the first question .Your statement contains fundamental errors. In the standard GAIL framework, expert demonstrations should be labeled as real data (label 1) and generated samples as fake data (label 0), which aligns perfectly with the adversarial logic of GANs. The generator aims to deceive the discriminator by maximizing its confidence in generated samples (pushing them toward 1), ​not by directly using ones() labels to enforce this assignment. Labeling expert samples as 0 would invert the discriminator's learning objective, contradict the generator's optimization direction in adversarial training, and ultimately undermine the core mechanism of imitation learning—progressively aligning the generated policy's distribution with the expert's. This inversion fatally disrupts the adversarial equilibrium required for effective policy imitation.our suggestions.

**Comment by cp15512516229** [comment]
Hello ！ About the first question .Your statement contains fundamental errors. In the standard GAIL framework, expert demonstrations should be labeled as real data (label 1) and generated samples as fake data (label 0), which aligns perfectly with the adversarial logic of GANs. The generator aims to deceive the discriminator by maximizing its confidence in generated samples (pushing them toward 1), ​not by directly using ones() labels to enforce this assignment. Labeling expert samples as 0 would invert the discriminator's learning objective, contradict the generator's optimization direction in adversarial training, and ultimately undermine the core mechanism of imitation learning—progressively aligning the generated policy's distribution with the expert's. This inversion fatally disrupts the adversarial equilibrium required for effective policy imitation.
